src/main/kotlin/com/example/delivery/service/ProductService.kt

```diff
@@ -27,4 +30,19 @@ class ProductService(private val productRepository: ProductRepository) {
     fun deleteById(id: String) {
         return productRepository.deleteById(id)
     }
+
+    fun update(id: String, updateProductDTO: UpdateProductDTO): DomainProduct {
+        return productRepository.update(id, createUpdateObject(updateProductDTO))?.toDomain()
+            ?: throw NotFoundException("Product with id $id doesn't exists")
+    }
+
+    private fun createUpdateObject(updateProductDTO: UpdateProductDTO): Update {
+        val update = Update()
+        with(updateProductDTO) {
+            name?.let { update.set(MongoProduct::name.name, this) }
+            price?.let { update.set(MongoProduct::price.name, this) }
+            amountAvailable?.let { update.set(MongoProduct::amountAvailable.name, this) }
```

Comment:
what is this here? where are tests to check that this code works correctly?

and same comment about SRP